### ISSUE TITLE
[Bug] - Cloud config - locale module is disabled

### ISSUE THREAD

**Issue by jeanblanchard** [amazonlinux/amazon-linux-2023#362]
**Describe the bug**
In AL2, the "locale" module was enabled in `/etc/cloud/cloud.cfg`, so you could easily set the default locale of an EC2 instance using the UserData

In AL2023, this is no longer the case, and the "locale" config is ignored

**To Reproduce**
Set the user-data of an AL2023-based EC2 instance to
```
#cloud-config
locale: fr_FR.UTF-8
```
On boot, the default locale is not set:
```
$ locale
LANG=C.UTF-8
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=
```

**Expected behavior**
The default locale is expected to be set to fr_FR, as it was in AL2:
```
$ locale
LANG=fr_FR.UTF-8
LC_CTYPE="fr_FR.UTF-8"
LC_NUMERIC="fr_FR.UTF-8"
LC_TIME="fr_FR.UTF-8"
LC_COLLATE="fr_FR.UTF-8"
LC_MONETARY="fr_FR.UTF-8"
LC_MESSAGES="fr_FR.UTF-8"
LC_PAPER="fr_FR.UTF-8"
LC_NAME="fr_FR.UTF-8"
LC_ADDRESS="fr_FR.UTF-8"
LC_TELEPHONE="fr_FR.UTF-8"
LC_MEASUREMENT="fr_FR.UTF-8"
LC_IDENTIFICATION="fr_FR.UTF-8"
LC_ALL=
```

**Workaround**
You can create a `/etc/cloud/cloud.cfg.d/30_locale.cfg` file to add the `locale` module to the ones from `/etc/cloud/cloud.cfg`:
```
cloud_config_modules:
 - set-passwords
 - yum-variables
 - yum-add-repo
 - ntp
 - locale
 - timezone
 - disable-ec2-metadata
 - runcmd
```
This is not ideal, since you then hardcode the whole list of config modules, it is not possible (to my knowledge) to just add a module.

**Comment by nmeyerhans** [comment]
The fix for this will be included in the next scheduled AL2023 release. ETA is approximately Aug. 22.

**Comment by w25r** [comment]
Any update on when the fix for this bug might be released?

**Comment by nmeyerhans** [comment]
Expect it within one of the next couple of releases, roughly by mid September.  I'd expected it to be available already but it was held up by an unrelated regression.

**Comment by nmeyerhans** [comment]
This should be fixed as of the 2023.2.20230920 release.

**Comment by lawrencekoenig** [comment]
Was this actually fixed in this release.  I cannot find mention of this in the release notes?
